.github/workflows/CI.yml

```
@@ -18,8 +18,17 @@ jobs:
         with:
           version: master
 
-      - name: testing
+      - name: testing (Linux)
+        if: startsWith(matrix.runs-on, 'ubuntu')
         run: sh runall.sh
+      
+      - name: testing (MacOS)
+        if: startsWith(matrix.runs-on, 'macos')
+        run: sh runall.sh
```

We can shorten the code a little bit here (untested).
```suggestion
      - name: testing (Linux)
        if: (startsWith(matrix.runs-on, 'ubuntu')) || (startsWith(matrix.runs-on, 'macos'))
        run: sh runall.sh
```